src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2012,7 +2017,7 @@ private void MarkFieldsUsed(TypeSymbol type)
                                 continue;
                             }
 
-                            FieldSymbol field = (FieldSymbol)symbol;
+                            FieldSymbol field = (FieldSymbol)symbol.OriginalDefinition;
```

Comment:
>FieldSymbol field = (FieldSymbol)symbol.OriginalDefinition; [](start = 28, length = 59)

It is better to adjust implementation of NoteFieldAccess. This change can actually have undesired effect on result of MarkFieldsUsed call below.